src/apple/macos/component/arm.rs

```diff
@@ -0,0 +1,120 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use std::ptr::null;
+use std::ffi::CStr;
+
+use core_foundation_sys::base::kCFAllocatorDefault;
+use core_foundation_sys::mach_port::CFIndex;
+use core_foundation_sys::base::Boolean;
+use core_foundation_sys::string::CFStringCreateWithBytes;
+use core_foundation_sys::string::CFStringGetCStringPtr;
+use core_foundation_sys::string::kCFStringEncodingUTF8;
+use core_foundation_sys::array::CFArrayGetCount;
+use core_foundation_sys::array::CFArrayGetValueAtIndex;
+
+use crate::ComponentExt;
+use crate::apple::inner::ffi::IOHIDEventSystemClientCopyServices;
+use crate::apple::inner::ffi::IOHIDEventSystemClientCreate;
+use crate::apple::inner::ffi::IOHIDEventSystemClientSetMatching;
+use crate::apple::inner::ffi::IOHIDServiceClientCopyProperty;
+use crate::apple::inner::ffi::IOHIDServiceClientRef;
+use crate::apple::inner::ffi::IOHIDServiceClientCopyEvent;
+use crate::apple::inner::ffi::IOHIDEventGetFloatValue;
+use crate::apple::inner::ffi::kHIDPage_AppleVendor;
+use crate::apple::inner::ffi::kHIDUsage_AppleVendor_TemperatureSensor;
+use crate::apple::inner::ffi::kIOHIDEventTypeTemperature;
+use crate::apple::inner::ffi::IOHIDEventFieldBase;
+use crate::apple::inner::ffi::matching;
+use crate::apple::inner::ffi::HID_DEVICE_PROPERTY_PRODUCT;
+
+pub(crate) fn temperatures() -> Vec<Component> {
+    let mut components = Vec::new();
+
+    unsafe {
+        let matches = matching(kHIDPage_AppleVendor, kHIDUsage_AppleVendor_TemperatureSensor);
+        let client = IOHIDEventSystemClientCreate(kCFAllocatorDefault);
+        let _ = IOHIDEventSystemClientSetMatching(client, matches);
+        let services = IOHIDEventSystemClientCopyServices(client);
+
+        let key_ref = CFStringCreateWithBytes(
+            kCFAllocatorDefault, 
+            HID_DEVICE_PROPERTY_PRODUCT.as_ptr(), 
+            HID_DEVICE_PROPERTY_PRODUCT.len() as CFIndex, 
```

Comment:
This would be simpler:

```suggestion
            HID_DEVICE_PROPERTY_PRODUCT.len() as _,
````